### DISCUSSION TITLE
打印功能

### DISCUSSION THREAD

**Discussion by yorkolee** [Tencent/tgfx#166] [Ideas] [answered]
请问一下什么时候可以实现打印功能？或者提供一个打印功能的思路，我将尝试先去实现。谢谢 ！

**Comment by domchen** [comment]
打印功能要支持转换绘制指令为pdf就行，需要扩展一个自定义Canvas，搜集指令都直接输出到pdf文件，目前有这块规划，但没有具体时间

**Reply by yorkolee** [reply]
如果是跨平台的话，用PDF就有字体方面的兼容问题，看来还是和skia cairo的思路相似，但这两个的效率有点低。如果把绘制指令转换为原生系统的绘制指令，相当于根据平台特性加一个封装层，这个有可行性吗？

**Reply by domchen** [reply]
PDF功能在上半年会提供。如果不希望有跨平台字体问题，可以直接渲染输出图片进行打印。

**Comment by domchen** [comment, marked as answer]
从 [v2.1](https://github.com/Tencent/tgfx/releases) 版本开始已经支持了PDF的导出。如果需要打印矢量，绘制后输出一个PDF对象发送给打印机即可。
